src/plugins/discover/public/application/main/hooks/use_text_based_query_language.ts

```diff
@@ -121,7 +123,7 @@ export function useTextBasedQueryLanguage({
               prev.current.columns = nextColumns;
             }
             const nextState = {
-              ...(addDataViewToState && { index: dataViewObj.id }),
+              ...(addDataViewToState && { index: undefined }),
```

Comment:
before this change, on every new index pattern entered in ES|QL, it's new id was propagated to URL, this is no longer necessary, which leads to less redundant URL state updates. The id of a ES|QL constructed ad-hoc data view is just relevant for internal usage, and is redundant in the URL. It just should be propagated to url as `undefined`, cleaning it up, when coming from a non-ES|QL query, where it's a requirement